### DISCUSSION TITLE
Add car from different account

### DISCUSSION THREAD

**Discussion by alexrusso51** [teslamate-org/teslamate#2563] [Q&A] [answered]
My wife just got a M3 with an account in her name/email. I am currently using TeslaMate with my MY. I want to add my wife's M3 to TeslaMate as well. Most importantly I want both to report as separate cars to my HomeAssistant MQTT broker. 

I'm running TeslaMate as a docker container in a Ubuntu VM on Proxmox. I spun up a new VM (RancehrOS) and got the new M3 tracking in there. I have them both pointed to my HA MQTT broker which is running as its own VM on the same Proxmox box. Problem is, they both report to MQTT with a car id of "1". So I can only see one of them at a time in MQTT/HA.

Any suggestions on how to fix this? 
Can I change the car_id for one of them? How? 
Can I somehow add the new car to my original TeslaMate install?

**Comment by cwanja** [comment, marked as answer]
Any reason you stood up its own TeslaMate instance for your wife Model 3? Your can run both inside of a single TeslaMate instance and each Tesla will have a unique `ID`. Data will stay unique and separate for both. In fact, if you ever have to log back into one or both instances, I imagine the other car is going to show up. Quite surprised when you setup the second instance and logged in, both cars did not show up.

>Can I change the car_id for one of them? How?

You could manually through the database, but do not advise it.
> Can I somehow add the new car to my original TeslaMate install?

You would need to be added to her Tesla account as a driver. Then it should show up in your Tesla app as a second car. Once confirmed, log out and log back into TeslaMate. TeslaMate should pickup the second car. 

Edit: re-read your first line - did not pickup you used her email address and password. See here: https://www.tesla.com/support/account-support#car-access "How can I add a driver for my vehicle?"

**Reply by alexrusso51** [reply]
My wife's car is on a different account with Tesla. My Tesla account is my email, hers is her email. They have different access tokens. How do I add my wife's car to my instance of TeslaMate?

I imagine since they are separate accounts only the one associated with the account the instance is logged in with shows up, and not the other.

**Reply by cwanja** [reply]
See some of my later edits, totally my fault for missing that they were separate accounts. You would need to be added as a driver to your wife's Tesla account (linked in original response). Once you do that, her Model 3 will show up in your Tesla app. Once you have done that, log out and back into TeslaMate and her Model 3 will show up and have its own `ID`. Probably 2.

**Reply by alexrusso51** [reply]
Thanks. Trying now. I'm already added as a driver for my wife's M3. Didn't think about logging out and in again. 

Logged out. Now having trouble logging in, keeps telling me tokens are invalid even though I just generated them with the iPhone app.

**Reply by alexrusso51** [reply]
Couldn't log in. Good old restart of the container did the trick. Everything working great now. The second car showed up. Thank you for your help!!!

**Reply by igeryu** [reply]
I am in a similar situation -- I have a non-primary vehicle show up in the API, for me, because I have an `access_type` of `DRIVER` (my spouse's vehicle).  I've noticed that some data that shows up for my own vehicle is missing for my spouse's vehicle.  Examples are

* **Efficiency** dashboard: "Consumption (net)" and "Consumption (gross)"
* **Trip** dashboard: "used" (positioned above "added (AC)", which _is_ populated)
* **Drive Stats** dashboard: "kWh used" and "Median kWh used per day"

For the Trip dashboard, even if I view previous days (thinking maybe the day must be "closed out"), the "used" is still blank.  All of those metrics are present for my own vehicle, with `access_type` of `OWNER`.

I'm sure there are others, but it seems like power consumption/usage is the core item not being recorded.
